### DISCUSSION TITLE
Delayed Disabling of HxDropdownToggleButton Allows Interaction and Causes Exception

### DISCUSSION THREAD

**Discussion by ppolyakov** [havit/Havit.Blazor#894] [General]
Hello,

Could you please clarify if the following behavior is correct:

Parent component:
The HxSwitch parameter controls the Enabled/Disabled state of the HxDropdownToggleButton component.
Example code:
```
<HxSwitch Label="HxSwitch" Text="switch this" @bind-Value="isEnabled" />
<HxDropdownComponent Enabled="isEnabled" />
```

Child component (HxDropdownComponent):
HxDropdownToggleButton with an Enabled parameter.
Example code:
```
<HxDropdownToggleButton Enabled="@Enabled" />
<HxDropdownMenu>
        <HxDropdownItem>Test1</HxDropdownItem>
        <HxDropdownItem>Test2</HxDropdownItem>
</HxDropdownMenu>
```

Due to a delay in server processing, the user can click on the HxDropdownToggleButton while the HxSwitch value has already changed to false, but the HxDropdownToggleButton hasn’t yet been disabled. As a result, the button can still be clicked, which opens the DropDown and leads to an exception.

_System.InvalidOperationException: The HxDropdownToggleButton component is in a disabled state.
   at void Havit.Diagnostics.Contracts.Contract.ThrowException<TException>(string message)
   at async Task Havit.Blazor.Components.Web.Bootstrap.HxButton.HandleClick(MouseEventArgs mouseEventArgs)
   at async Task Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at async Task Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)_

In the Havit source code (line 240 throws an exception):
![image](https://github.com/user-attachments/assets/f943a6ba-4815-448b-bdcc-c6cd228b75fd)

Thank you

**Comment by hakenr** [comment]
Thanks for reporting this. We will most likely have to change the line 240 not to throw exception but to ignore the click. I will create an issue to track the progress.
